module/VuFind/src/VuFind/View/Helper/Root/DisplayLanguageOptionFactory.php

```diff
@@ -65,10 +65,9 @@ public function __invoke(ContainerInterface $container, $requestedName,
         if (!empty($options)) {
             throw new \Exception('Unexpected options sent to factory.');
         }
-        // We want to construct a separate translator instance for this helper,
-        // since it configures different language/locale than the core shared
-        // instance!
-        $factory = new \VuFind\I18n\Translator\TranslatorFactory();
-        return new $requestedName($factory($container, Translator::class));
+        $translator = $container->get(Translator::class);
+        // Add a special locale used just for this plugin:
+        $translator->addTranslationFile('ExtendedIni', null, 'default', 'native');
```

Comment:
Perhaps it makes more sense to put this in the actual TranslatorFactory; I'm torn between doing that (which is probably more straightforward) and leaving it here (since it ties the loading of the translations to the use of the helper that needs them). Any thoughts/preferences, @EreMaijala?

Reply:
@demiankatz I like it the way it is now. As you said it ties the logic with the helper that needs it.